department-of-veterans-affairs/abd-vro
# Introduces a metric_logger ruby implementation for use in the bgs sevice

Adds a request count metric for requests started, and requests completed, and a request latency metric for calls out to the BGS service.

## What was the problem?
We had extremely limited visibility into the goings on with the external calls that the VRO svc-bgs-api client service makes. 

Associated tickets or Slack threads:
- #2980

## How does this fix it?[^1]
With this PR, we'll be able to see request start, complete counts as well as tracking the request latency.

## How to test this PR
- Running the bgs integration test will now submit metrics via the datadog api.


[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
